libs/src/api/serviceProvider.js

```diff
@@ -96,17 +96,25 @@ class ServiceProvider extends Base {
       .filter(Boolean)
 
     // Time requests and autoselect nodes
-    const timings = await timeRequestsAndSortByVersion(
+    let timings = await timeRequestsAndSortByVersion(
       creatorNodes.map(node => ({
         id: node,
         url: `${node}/health_check`
       }))
     )
 
+    // Store all the healthy services in a map. Used in UI to select other available services
     let services = {}
+    const available = []
     timings.forEach(timing => {
-      services[timing.request.id] = timing.response.data
+      // Filter out unhealthy nodes (nodes with no response)
+      if (timing.response) {
+        services[timing.request.id] = timing.response.data
+        available.push(timing)
+      }
```

Comment:
This seems fine. I think this all is a bit more convoluted than it needs to be. You could just do

`timeRequests`
```
try {
    response = await axios.get(request.url)
  } catch (e) {
    console.debug(`Error with request for ${request.url}: ${e}`)
    return null
  }
```

`autoselectCreatorNodes`
```
timings.filter(Boolean).forEach(timing => ...))

up to you